src/utils.c

```diff
@@ -207,3 +207,52 @@ get_envvar_filename(guint port)
     }
     return(filename);
 }
+
+/* get_install_dir()
+ *
+ * Get directory to install tasks by configured file or default.
+ *
+ * If there was an unexpected result in getting data from
+ * configured install_config file, an error will be returned
+ * with default_install_path.
+ */
+gchar *
+get_install_dir(GError **error)
+{
+    gchar *filename;
+    GKeyFile *keyfile = NULL;
+    gchar *install_dir_value;
+
+    /* If the expected directory is present, this is an actual installation */
+    if (g_file_test("/var/lib/restraint", G_FILE_TEST_IS_DIR)) {
+        filename = g_strdup_printf("%s/%s", "/var/lib/restraint", INSTALL_CONFIG_FILE);
+    } else {
+        filename = g_strdup_printf("%s/%s", ".", INSTALL_CONFIG_FILE);
+    }
```

Comment:
You don't need this for testing. The purpose of the function should be to get the value from the configuration file,
```
gchar *get_install_dir (const gchar *config_file, GError **error);
```
Return NULL if there is no value or there is an error. The cases where the file is missing or the key is not in the file are valid, so no error should be set for these.
Error case should be that the file is present but cannot be read, these should be covered already by g_key_file.
What happens if the key is in the file without a value?

In the caller, recipe.c, check return value, if it's not NULL, use it. If it's NULL and error is set report it.
If it's NULL, and there is no error, just take the value from the default, which was already defined in TASK_LOCATION macro, so you can keep it and update the value.

The macro INSTALL_CONFIG_FILE can be already,
```
#define INSTALL_CONFIG_FILE "/var/lib/restraint/install_config"
```
Should be in recipe, and it's the config_file argument for the call. Then in tests, pass whatever file you have for tests.